### PR TITLE
Add introduction paragraph to README

### PR DESCRIPTION
**Pull request by RyanZim**
This is an attempt at fixing #138 & resolving #148.

Not sure if I'm entirely satisfied with it; suggestions welcome. @RobLoach?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 <a href="https://coveralls.io/r/jstransformers/jstransformer?branch=master"><img src="https://img.shields.io/coveralls/jstransformers/jstransformer/master.svg" alt="Coverage Status"></a>
 <a href="https://www.npmjs.org/package/jstransformer"><img src="https://img.shields.io/npm/v/jstransformer.svg" alt="NPM version"></a></p>
 
+There are many good template engines, compilers, and transpilers written for node.js. But there is a problem: all of them have slightly different APIs, requiring slightly different usage. JSTransformer unifies them into one standardized API. Code written for one transformer will work with any other transformer. There are over 100 transformers, ranging from markdown parsers to template engines to coffeescript transpilers.
```
Comment (TimothyGu) [inline README.md]: You might want to link the [jstransformer keyword](https://www.npmjs.com/browse/keyword/jstransformer). Also capitalize Node.js, Markdown, and CoffeeScript.

Is it necessary to write "compilers and transpilers"? They are pretty synonymous from a JavaScript standpoint.

Also, I mean CoffeeScript isn't really that popular anymore. Maybe Babel?

Comment (RyanZim) [inline README.md]: > Is it necessary to write "compilers and transpilers"?

Perhaps not.

BTW, what is the proper term for markdown-to-html tools? Parsers? Compilers? Renderers?

> CoffeeScript isn't really that popular anymore. Maybe Babel?

Good point, IDK, just something that shows the diversity of JSTransformers.

Will definitely link to the keyword; thanks for the capitalization notes.

**Comment by RyanZim** [comment]
@TimothyGu I make some updates.

**Comment by RyanZim** [comment]
@TimothyGu @RobLoach ping?

**Pull request merged**